File: backend/event-microservice/Event/Event.Application/Dto/EventDetailsDto.cs
using Event.Domain.Entities;

namespace Event.Application.Dto;

public class EventDetailsDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public EventCategory Category { get; set; }
    public DateTimeOffset FromDate { get; set; }
    public DateTimeOffset ToDate { get; set; }
    public EventAddressDto Address { get; set; } = null!;
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
    public bool IsPrivate { get; set; }
}

File: frontend/event_planr/lib/data/network/models/event_details_dto.dart
import 'package:event_planr/data/network/models/event_address_dto.dart';
import 'package:event_planr/utils/utils.dart';
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';

part 'event_details_dto.g.dart';

@JsonSerializable()
@DateTimeConverter()
@immutable
class EventDetailsDto {
  const EventDetailsDto({
    required this.id,
    required this.name,
    required this.category,
    required this.fromDate,
    required this.toDate,
    required this.address,
    required this.description,
    required this.coverImageUrl,
    required this.isPrivate,
  });

  factory EventDetailsDto.fromJson(Map<String, dynamic> json) =>
      _$EventDetailsDtoFromJson(json);

  Map<String, dynamic> toJson() => _$EventDetailsDtoToJson(this);

  final String id;
  final String name;
  final int category;
  final DateTime fromDate;
  final DateTime toDate;
  final EventAddressDto address;
  final String? description;
  final String? coverImageUrl;
  final bool isPrivate;
}
